include/boost/multiprecision/cpp_int/cpp_int_config.hpp

```diff
@@ -17,57 +21,80 @@ namespace detail {
 
 //
 // These traits calculate the largest type in the list
-// [unsigned] boost::long_long_type, long, int, which has the specified number
-// of bits.  Note that intN_t and boost::int_t<N> find the first
+// [unsigned] long long, long, int, which has the specified number
+// of bits.  Note that int_t and uint_t find the first
 // member of the above list, not the last.  We want the last in the
 // list to ensure that mixed arithmetic operations are as efficient
 // as possible.
 //
+
+template <unsigned Bits>
+struct int_t
+{
+   using exact = typename std::conditional<Bits == sizeof(signed char) * CHAR_BIT, signed char,
+                 typename std::conditional<Bits == sizeof(short) * CHAR_BIT, short,
+                 typename std::conditional<Bits == sizeof(int) * CHAR_BIT, int,
+                 typename std::conditional<Bits == sizeof(long) * CHAR_BIT, long, long long
```

Comment:
Out of an over-abundance of caution, there's just a couple more things we should do here I think.

In this case, before setting `type` to `[unsigned] long long`, we should check that the bit count matches up, in the case that we have no matching type, maybe the best we can do is set exact to to type `void`, any better ideas?